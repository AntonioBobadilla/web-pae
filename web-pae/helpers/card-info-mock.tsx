import { History } from '../components/card-info/types';

const history: History[] = [
  {
    date: 'Lunes 21 de Marzo - 3:00 PM - 4:00 PM',
    subject: 'Programaci贸n orientada a objetos',
    student: 'Karen Rugerio Armenta',
    status: 'pending'
  },
  {
    date: 'Lunes 21 de Marzo - 3:00 PM - 4:00 PM',
    subject: 'Programaci贸n orientada a objetos',
    student: 'Karen Rugerio Armenta',
    status: 'pending'
  },
  {
    date: 'Lunes 21 de Marzo - 3:00 PM - 4:00 PM',
    subject: 'Programaci贸n orientada a objetos',
    student: 'Karen Rugerio Armenta',
    status: 'pending'
  },
  {
    date: 'Lunes 21 de Marzo - 3:00 PM - 4:00 PM',
    subject: 'Programaci贸n orientada a objetos',
    student: 'Karen Rugerio Armenta',
    status: 'pending'
  }
];
// eslint-disable-next-line import/prefer-default-export
export function HistoryMockService() {
  return history;
}
