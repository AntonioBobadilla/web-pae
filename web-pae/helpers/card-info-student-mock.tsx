import { Tutoring } from '../components/card-info-student/types';

const historystu: Tutoring[] = [
  {
    date: 'Viernes 27 de Mayo - 9:00 AM - 10:00 AM',
    subject: 'Programación Orientada a Objetos',
    student: 'Bryan González Arellano',
    location: 'https://itesm.zoom.us/j/5519309007',
    status: 'pending'
  },
  {
    date: 'Lunes 27 de Marzo - 3:00 PM - 4:00 PM',
    subject: 'Estructuras simbólicas en la imagen, la literatura y la música',
    student: 'Karen Rugerio Armenta',
    location: 'A1332',
    status: 'info'
  }
];
// eslint-disable-next-line import/prefer-default-export
export function HistoryStuMockService() {
  return historystu;
}
