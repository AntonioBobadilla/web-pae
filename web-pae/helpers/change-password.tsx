export type ModifyPasswordData = {
  currentPassword: string;
  newPassword: string;
  passwordConfirmation: string;
};
async function changePassword(data: ModifyPasswordData, token: string) {
  try {
    const response = await fetch(
      'https://server-pae.azurewebsites.net/changepassword/',
      {
        method: 'PATCH',
        headers: {
          'Content-Type': 'application/json',
          Authorization: `Token ${token}`
        },
        body: JSON.stringify({
          password: data.currentPassword,
          new_password: data.newPassword,
          confirm_new_password: data.passwordConfirmation
        }),
        cache: 'no-cache',
        credentials: 'same-origin'
        // mode: 'cors'
      }
    );
    const status = await response.status;
    const responseData = await response.json();

    return { status, responseData };
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
  } catch (err: any) {
    // console.log(err.message);
    return { status: 500, message: err.message };
    // sessionStorage.setItem('token', JSON.stringify(userToken));
  }
}

export default changePassword;
