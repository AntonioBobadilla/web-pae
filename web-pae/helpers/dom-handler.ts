export default class DomHandler {
  static hasDOM() {
    return !!(
      typeof window !== 'undefined' &&
      window.document &&
      window.document.createElement
    );
  }

  static getBrowserLanguage() {
    return (
      navigator.language ||
      (navigator.languages &&
        navigator.languages.length &&
        navigator.languages[0]) ||
      navigator.language ||
      navigator.language ||
      navigator.language ||
      'en'
    );
  }
}
