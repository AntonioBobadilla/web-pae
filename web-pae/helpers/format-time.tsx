function FormatTime(time: number) {
  return `${time.toString()}:00 - ${(time + 1).toString()}:00`;
}
export default FormatTime;
