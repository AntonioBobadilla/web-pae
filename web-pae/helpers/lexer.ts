/* eslint-disable no-console */
/* eslint-disable @typescript-eslint/no-explicit-any */
class Lexer {
  tokenRegExp: any = undefined;

  // Perform validation and freeze rules so there are no side-effects.
  rules: any = [];

  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
  constructor(newRules: any) {
    if (!newRules || !newRules.length) {
      console.log('Rules must be of type Array.');
    } else {
      for (let i = 0, ii = newRules.length; i < ii; i += 1) {
        const rule = newRules[i];
        if (!rule || !rule.length || rule.length > 3) {
          console.log(`Invalid rule at index ${i}.`);
        }
        let expression = rule[0];
        if (!(expression instanceof RegExp || expression instanceof String)) {
          console.log(
            `${
              'Expression must be an instance of RegExp or String' +
              ' for rule at index '
            }${i}.`
          );
        }
        // Prevent side-effects by taking string copy of RegExp. This also has
        // the benefit of stripping all modifiers from the RegExp.
        if (expression.source) {
          expression = expression.source;
        } else {
          expression = expression.replace(/[-*+?.,^$|#[\]{}()\\]/g, '\\$1');
        }

        const type = rule[1];
        if ((typeof type !== 'string' || type.length === 0) && type !== null) {
          console.log(
            `Expected String or null instead found ${JSON.stringify(
              String(type)
            )} for type of rule at index ${i}.`
          );
        }
        const action = rule[2];
        if (action !== undefined && !(action instanceof Function)) {
          console.log(
            `${
              'Constructor is defined, but is not a function for ' +
              'rule at index '
            }${i}.`
          );
        }
        this.rules[i] = [expression, type, action];
      }
    }
  }

  compile(): any {
    if (!this.tokenRegExp) {
      const tokenExpressions = [];
      const { rules } = this;
      let captureCount = 0;
      for (let i = 0, ii = rules.length; i < ii; i += 1) {
        const rule = rules[i];
        let expression = rule[0];

        // How many captures does this expression have?
        const CAPTURE_EXP = /\\.|\[(?:\\.|[^\]])*\]|(\((?!\?[!:=]))|./g;
        const captures = expression.replace(CAPTURE_EXP, (match: any, p: any) =>
          p ? '.' : ''
        ).length;
        rule[3] = captures + 1;

        // Increment backreferences.
        const BACK_REF_EXP = /\\\D|\[(?:\\.|[^\]])*\]|\\(\d+)|./g;
        expression = expression.replace(
          BACK_REF_EXP,
          // eslint-disable-next-line no-loop-func
          (match: any, d: string) => {
            if (d) {
              const n = parseInt(d, 10);
              if (n > 0 && n <= captures) {
                return `\\${n + captureCount + 1}`;
              }
              return parseInt(d, 8); // Assume this was an escape sequence?
            }
            return match;
          }
        );

        captureCount += captures + 1;

        tokenExpressions.push(expression);
      }
      this.tokenRegExp = new RegExp(`(${tokenExpressions.join(')|(')})`, 'g');
    }

    return this.tokenRegExp;
  }

  lex(text: string): any {
    if (typeof text !== 'string') {
      console.log('Attempt to lex an Object that is not a String.');
    }

    const tokens = [];
    let tokenMatch;
    const tokenRegExp = this.compile();

    // eslint-disable-next-line no-cond-assign
    while ((tokenMatch = tokenRegExp.exec(text))) {
      // Throw if character is skipped.
      const token = {
        type: '',
        value: tokenMatch[0]
      };

      // Do a linear search for the group that matched then look up its
      // corresponding token.
      let i = 1;
      let r = 0;
      const { rules } = this;
      let rule = rules[r];
      while (!tokenMatch[i]) {
        i += rule[3];
        // eslint-disable-next-line no-plusplus
        rule = rules[++r];
      }
      // eslint-disable-next-line prefer-destructuring
      token.type = rule[1];

      // Throw an exception rather than enter an infinite loop.
      if (tokenMatch[0].length === 0) {
        console.log(`Rule at index ${i} matched the empty string.`);
      }

      // If the type is null then token will be thrown away.
      if (token.type) {
        tokens.push(token);
      }
      //   console.log(token);
    }

    return tokens;
  }
}

export default Lexer;
