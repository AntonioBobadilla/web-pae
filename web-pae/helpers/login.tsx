const login = async (
  obj: { username: string; password: string },
  url: string
) => {
  let auth = true;
  try {
    const response = await fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(obj)
    });
    const status = await response.status;
    const json = await response.json();
    const ok = await response.ok;

    const { message } = json;

    if (!ok) {
      auth = false;
    }

    return { auth, message };
  } catch (err: any) {
    auth = false;
    // console.log(err.message);
    return { auth, message: err.message };
  }
};

export default login;
