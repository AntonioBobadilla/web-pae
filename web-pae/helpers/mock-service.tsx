import { Meeting } from '../components/data-table/types';

const meetings: Meeting[] = [
  {
    isOnline: false,
    date: '2020-05-05',
    hour: 11
  },
  {
    isOnline: true,
    date: '2020-05-06',
    hour: 12
  },
  {
    isOnline: true,
    date: '2020-05-07',
    hour: 13
  },
  {
    isOnline: true,
    date: '2020-05-08',
    hour: 14
  },
  {
    isOnline: true,
    date: '2020-05-09',
    hour: 15
  },
  {
    isOnline: true,
    date: '2020-05-10',
    hour: 16
  }
];
// eslint-disable-next-line import/prefer-default-export
export function MeetingsMockService() {
  return meetings;
}
