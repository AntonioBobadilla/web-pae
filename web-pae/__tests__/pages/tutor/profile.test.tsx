import CardInfo from '@/components/card-info';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { HistoryMockService } from 'helpers/card-info-mock';
import React from 'react';

describe('Tutor Profile', () => {
  it('renders tutor profile page unchanged', () => {
    const mockService = HistoryMockService();
    const { container } = render(
      <>
        {mockService.map((historyItem) => (
          <CardInfo
            date={historyItem.date}
            subject={historyItem.subject}
            student={historyItem.student}
            status={historyItem.status}
            key={historyItem.date}
          />
        ))}
      </>
    );
    expect(container).toMatchSnapshot();
  });
});
