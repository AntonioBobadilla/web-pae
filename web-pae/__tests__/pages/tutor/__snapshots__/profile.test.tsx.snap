// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tutor Profile renders tutor profile page unchanged 1`] = `
<div>
  <div>
    <h1>
      PROFILE
    </h1>
    <div
      class="cardInfo"
    >
      <div
        class="box"
      >
        <div
          class="dateSize"
        >
          <h1
            class="bi bi-alarm icon"
          >
             
          </h1>
          <span
            class="caption"
          >
             
            Lunes 21 de Marzo - 3:00 PM - 4:00 PM
             
          </span>
        </div>
        <div
          class="subjectSize"
        >
          <h1
            class="bi bi-stack icon"
          >
             
          </h1>
          <span
            class="caption"
          >
             
            Programaci√≥n orientada a objetos
             
          </span>
        </div>
        <div
          class="studentSize"
        >
          <h1
            class="bi bi-person-circle icon"
          >
             
          </h1>
          <span
            class="caption"
          >
             
            Karen Rugerio Armenta
             
          </span>
        </div>
        <div
          class="statusSize"
        >
          <button
            class="container button info"
          >
             
            Completada 
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
